File: mobile/android/OpenU60/app/src/main/java/com/openu60/core/network/AgentError.kt
package com.openu60.core.network

sealed class AgentError(override val message: String, override val cause: Throwable? = null) : Exception(message, cause) {
    class Unauthorized : AgentError("Not authenticated. Please log in.")
    class ServerError(msg: String) : AgentError("Server error: $msg")
    class NetworkError(msg: String, cause: Throwable? = null) : AgentError("Network error: $msg", cause)
    class DecodingError(msg: String) : AgentError("Failed to decode response: $msg")
    class ServerUnreachable : AgentError("Cannot reach the agent")
    class Timeout : AgentError("Request timed out")
}

File: mobile/ios/OpenU60/Core/Networking/AgentError.swift
import Foundation

enum AgentError: LocalizedError {
    case unauthorized
    case serverError(String)
    case networkError(Error)
    case decodingError(String)
    case serverUnreachable
    case timeout

    var errorDescription: String? {
        switch self {
        case .unauthorized:
            return "Not authenticated. Please log in."
        case .serverError(let message):
            return "Server error: \(message)"
        case .networkError(let error):
            return "Network error: \(error.localizedDescription)"
        case .decodingError(let detail):
            return "Failed to decode response: \(detail)"
        case .serverUnreachable:
            return "Cannot reach the agent"
        case .timeout:
            return "Request timed out"
        }
    }
}
